configure

```diff
@@ -1612,254 +1609,141 @@ case "${ARCH}" in
-                        SFLAGS="${SFLAGS} -DARM_ACLE"
-
-                        ARCH_STATIC_OBJS="${ARCH_STATIC_OBJS} crc32_acle.o insert_string_acle.o"
-                        ARCH_SHARED_OBJS="${ARCH_SHARED_OBJS} crc32_acle.lo insert_string_acle.lo"
-                    fi
-
-                    if test $buildneon -eq 1; then
-                        CFLAGS="${CFLAGS} -DARM_NEON"
-                        SFLAGS="${SFLAGS} -DARM_NEON"
-
-                        if test $MFPU_NEON_AVAILABLE -eq 1;then
-                            neonflag="-mfpu=neon"
-                        fi
-
-                        if test $NEON_HAS_LD4 -eq 1; then
-                            CFLAGS="${CFLAGS} -DARM_NEON_HASLD4"
-                            SFLAGS="${SFLAGS} -DARM_NEON_HASLD4"
-                        fi
-
-                        ARCH_STATIC_OBJS="${ARCH_STATIC_OBJS} adler32_neon.o chunkset_neon.o compare256_neon.o slide_hash_neon.o"
-                        ARCH_SHARED_OBJS="${ARCH_SHARED_OBJS} adler32_neon.lo chunkset_neon.lo compare256_neon.lo slide_hash_neon.lo"
-                    fi
-                fi
-            ;;
-        esac
-
-    ;;
-    # 64-bit ARM specific optimizations
-    aarch64)
-        [ ! -z $CROSS_PREFIX ] && QEMU_ARCH=aarch64
-        ARCHDIR=arch/arm
-
-        ARCH="armv8-a"
-
-        if test $without_optimizations -eq 0; then
-            check_neon_ld4_intrinsics
-
-            CFLAGS="${CFLAGS} -DARM_FEATURES"
-            SFLAGS="${SFLAGS} -DARM_FEATURES"
-            ARCH_STATIC_OBJS="${ARCH_STATIC_OBJS} arm_features.o"
-            ARCH_SHARED_OBJS="${ARCH_SHARED_OBJS} arm_features.lo"
-
-            if test $LINUX -eq 1; then
-                cat > $test.c <<EOF
+                        cat > $test.c <<EOF
 #include <sys/auxv.h>
 int main() {
-    return (getauxval(AT_HWCAP) & HWCAP_CRC32);
+    return (getauxval(AT_HWCAP) & HWCAP_NEON);
 }
 EOF
-                if try $CC -c $CFLAGS $test.c; then
-                    CFLAGS="${CFLAGS} -DARM_AUXV_HAS_CRC32"
-                    SFLAGS="${SFLAGS} -DARM_AUXV_HAS_CRC32"
-                else
-                    echo "HWCAP_CRC32 not present in sys/auxv.h; cannot detect support at runtime." | tee -a configure.log
+                        if try $CC -c $CFLAGS $test.c; then
+                            CFLAGS="${CFLAGS} -DARM_AUXV_HAS_NEON"
+                            SFLAGS="${SFLAGS} -DARM_AUXV_HAS_NEON"
+                        else
+                            echo "Neither HWCAP_ARM_NEON or HWCAP_NEON present in sys/auxv.h; cannot detect support at runtime." | tee -a configure.log
+                        fi
+                    fi
                 fi
             fi
 
-            if test $NEON_HAS_LD4 -eq 1; then
-                CFLAGS="${CFLAGS} -DARM_NEON_HASLD4"
-                SFLAGS="${SFLAGS} -DARM_NEON_HASLD4"
-            fi
+            check_acle_compiler_flag
 
-            if test $buildacle -eq 1; then
```

Comment:
This check looks like it should be kept so that building with `--without-acle` still works as expected.

Reply:
Good catch.